### PR TITLE
ResultEntitySource interface + CsvResultEntitySource impl

### LINKED ISSUE #388: Add support for ResultEntity Source
Currently,  I/O operations for `ResultEntity` instances is only supported in sinks, but no sources are available. An interface and potentially an implementation to read `ResultEntity` instances from e.g. a .csv-Source would be very helpful.

### PR DESCRIPTION
**Pull request by johanneshiry**
Resolves #388

### CODE REVIEW

**Comment by johanneshiry** [comment]
!test

File: src/main/java/edu/ie3/datamodel/io/source/csv/CsvDataSource.java [hunk 1]
```diff
@@ -620,4 +622,17 @@ protected void logSkippingWarning(
     return nodeAssetInputEntityDataStream(assetInputEntityDataStream(entityClass, operators), nodes)
         .map(dataOpt -> dataOpt.flatMap(factory::get));
   }
+
+  /**
+   * TODO JH
```
Comment (baszach) [inline src/main/java/edu/ie3/datamodel/io/source/csv/CsvDataSource.java:627]: Can we remove this TODO?

Comment (baszach) [reply src/main/java/edu/ie3/datamodel/io/source/csv/CsvDataSource.java]: Removed TODO and added java docs.

**Comment by baszach** [comment]
@johanneshiry do you think we can approve and merge this PR? Please, take a look at the test I wrote (`CsvResultEntitySourceTest`) and tell me if it needs changing/additions.

**Review by baszach — APPROVED** [review]

**Pull request merged**